code/modules/clothing/clothing_accessories.dm

```diff
@@ -32,6 +32,16 @@
 
 	..()
 
+/**
+ *  Attach accessory A to src
+ *
+ *  user is the user doing the attaching. Can be null, such as when attaching
+ *  items on spawn
+ */
+/obj/item/clothing/proc/attach_accessory(mob/user, obj/item/clothing/accessory/A)
+	accessories += A
+	A.on_attached(src, user)
```

Comment:
`/obj/item/clothing/proc/removetie_verb` adding is missed
And icon should be updated.

Reply:
fixed